synfig-studio/src/gui/duckmatic.cpp

```diff
@@ -2992,8 +2994,50 @@ Duckmatic::add_to_ducks(const synfigapp::ValueDesc& value_desc,etl::handle<Canva
                                                                      0.0f),             // location
                                                           value_desc));                 // value_desc
             add_duck(duck);
+            bezier->p1 = bezier->c1 = duck;
         }
 
+        //parent tip fake
+	    {
+	    	if(!bone.is_root()){
+			    synfigapp::ValueDesc parent_value_desc(bone_value_node, bone_value_node->get_link_index_from_name("parent"), orig_value_desc);;
+			    Bone parent_bone = (*(bone.get_parent()))(time).get(Bone());
+
+			    if(parent_bone.get_parent()){
+				    Point origin = parent_bone.get_origin();
+				    Matrix bm = parent_bone.get_animated_matrix();
+				    Real bscale = parent_bone.get_scalelx();
+
+				    parent_bone = (*parent_bone.get_parent())(time).get(Bone());
+				    Matrix m = parent_bone.get_animated_matrix();
+				    Real scale = parent_bone.get_scalelx();
+				    origin[0]*=scale;
+
+				    //Changing parent's origin to global coordinates
+				    origin = m.get_transformed(origin);
+
+				    //Obtaining Tip's location in global coordinates
+					origin += Point(bscale*bm.axis(0)[0],bscale*bm.axis(0)[1]);
+
+					//Converting tip's coordinates into local bone coordinates 
+				    m = bone.get_animated_matrix().get_inverted();
+				    origin = m.get_transformed(origin);
+
+				    synfigapp::ValueDesc value_desc(bone_value_node, bone_value_node->get_link_index_from_name(recursive ? "scalex" : "scalelx"), orig_value_desc);
```

Comment:
This line introduces LGTM warning - https://lgtm.com/projects/g/synfig/synfig/rev/pr-0895b82cba7353ae5ed4fae8e19ccf6a405b652f
Please fix.